### PR TITLE
SAIC-169 Key pair migration

### PR DESCRIPTION
**Pull request by roman-verchikov**
Key pairs are only accessible by user created them, admin user has no
access to user key pairs. Because of that migration fails for non-admin
VMs which have keypair associated.

### CODE REVIEW

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.

File: cloudferrylib/os/actions/transport_compute_resources.py [hunk 1]
```diff
@@ -38,3 +40,144 @@ def run(self, info=None, identity_info=None, **kwargs):
...
+            dst_nova.mysql_connector.execute(sql)
+
+    def run(self, **kwargs):
+        src_keystone = self.src_cloud.resources[utl.IDENTITY_RESOURCE]
+        dst_keystone = self.dst_cloud.resources[utl.IDENTITY_RESOURCE]
+
+        LOG.info('Fetching users present on SRC and DST, this may take awhile')
+
+        src_users = src_keystone.get_users_list()
+        dst_users = dst_keystone.get_users_list()
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/transport_compute_resources.py]: It's definitely a very bad idea to fetch all users, there can be hundreds of thousands of them in corporate LDAP in general case. I will rewrite this to only get user ID which has key pair on SRC

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.

**Comment by roman-verchikov** [comment]
Ok, so the problem is actually more complicated than I thought. Simply put, it's not enough to transfer key pairs to destination on the low level, it is also required to boot VM on destination with original user credentials, in our case VM is booted with admin credentials at all times, which makes it impossible to associate VM with user's key pair (key pair is only visible to user, not to admin). I will continue investigating the possibility to hack cloud-init on the lower level so that it provides correct keys at boot time without `--key-pair` argument provided to `nova boot`.

**Comment by Wasapl** [comment]
Merged build finished. Test FAILed.

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.

**Comment by plomakin** [comment]
+1 Works good for simple/devlab scenarios

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.

**Pull request merged**